### DISCUSSION TITLE
Superelements

### DISCUSSION THREAD

**Discussion by pan-zhuo** [uwlib-cams/sinopia_maps#88] [decided]
Superelements are ones composed of two or more subelements.

Examples

**publication statement** consists of **date of publication**, **name of publisher**, and **place of publication**

**dissertation or thesis information** consists of **academic degree**, **degree granting institution**, **year degree granted**

---
Do we want to include a superelement if all of its subelements are implemented in a resource template (RT)?

Reasons for including:
- Data sets converted from legacy formats may include these elements. We may want the RTs to load and display them properly.
- You could have 'order' or 'pairing' of values with superelements using a string encoding scheme. For example, two places of publication with two names of publishers, which goes with which?
- Catalogers can always choose to ignore if they wish not to record them.

Reasons for not including:
- It might be confusing to have superelements and subelements together. The element list in an RT is already long enough.
- 'Order' and 'pairing' of values are optional, not required.

**Comment by briesenberg07** [comment]
I think the strongest reason for including superelements like those you mention is:

> Data sets converted from legacy formats may include these elements [preserved as single values and not mapped to multiple subelements]. We may want the RTs to load and display them properly.

If MARC2RDA mapping indicates that publication statements, for example, will be output as single values for the publication statement superelement, this seems a reason to include it.

I also noted:

> You could have 'order' or 'pairing' of values with superelements using a string encoding scheme. For example, two places of publication with two names of publishers, which goes with which?

Sinopia property templates which are repeatable may now be specified as 'ordered' (http://sinopia.io/vocabulary/propertyAttribute/ordered). Providing multiple, ordered values for two properties *could* provide a way to associate publisher names with locations, for example. As you indicate, superelements might provide another.

**Reply by briesenberg07** [reply]
> Providing multiple, ordered values for two properties could provide a way to associate publisher names with locations, for example. As you indicate, superelements might provide another.


BUT, one publisher can be associated with more than one place.

**Comment by briesenberg07** [comment]
**📢 DECISION 2023-03-03**

In general, implement superelements. These will be used for converting MARC21 to RDA/RDF, and ideally RDA/RDF templates will accommodate this output data.
